src/test.rs

```diff
@@ -0,0 +1,199 @@
+wasmtime::component::bindgen!({
+    path: "wit",
+    world: "fermyon:spin/redis-trigger",
+    async: true
+});
+
+use {
+    anyhow::{anyhow, bail, Context, Result},
+    http_body_util::{combinators::BoxBody, BodyExt, Empty},
+    hyper::Request,
+    std::{ops::Deref, sync::OnceLock},
+    tokio::{
+        fs,
+        process::Command,
+        sync::{oneshot, OnceCell},
+        task,
+    },
+    wasmtime::{
+        component::{Component, Linker, ResourceTable},
+        Config, Engine, Store,
+    },
+    wasmtime_wasi::preview2::{WasiCtx, WasiCtxBuilder, WasiView},
+    wasmtime_wasi_http::{WasiHttpCtx, WasiHttpView},
+    wit_component::ComponentEncoder,
+};
+
+struct Ctx {
+    table: ResourceTable,
+    wasi: WasiCtx,
+    wasi_http: WasiHttpCtx,
+}
+
+impl WasiHttpView for Ctx {
+    fn ctx(&mut self) -> &mut WasiHttpCtx {
+        &mut self.wasi_http
+    }
+
+    fn table(&mut self) -> &mut ResourceTable {
+        &mut self.table
+    }
+}
+
+impl WasiView for Ctx {
+    fn ctx(&mut self) -> &mut WasiCtx {
+        &mut self.wasi
+    }
+
+    fn table(&mut self) -> &mut ResourceTable {
+        &mut self.table
+    }
+}
+
+async fn build_component(name: &str) -> Result<Vec<u8>> {
+    static BUILD: OnceCell<()> = OnceCell::const_new();
+
+    BUILD
+        .get_or_init(|| async {
+            assert!(
+                Command::new("cargo")
+                    .current_dir("test-cases")
+                    .args(["build", "--workspace", "--target", "wasm32-wasi"])
+                    .status()
+                    .await
+                    .unwrap()
+                    .success(),
+                "cargo build failed"
+            );
+        })
+        .await;
+
+    const ADAPTER_PATH: &str = "target/wasi_snapshot_preview1.reactor.wasm";
+
+    static GET_ADAPTER: OnceCell<()> = OnceCell::const_new();
+
+    GET_ADAPTER
```

Comment:
I personally think we should just check this in rather than relying on fetching it from the network.